src/App.tsx

```diff
@@ -15,48 +15,105 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((prev, next) => {
+    switch (sortType) {
+      case SortType.LENGTH:
+        return prev.length - next.length;
+
+      case SortType.ALPHABET:
+        return prev.localeCompare(next);
+
+      case SortType.NONE:
```

Comment:
don't add an empty case in switch case, because you have a default case for that